dpnp/linalg/linalg_iface.py

```diff
@@ -125,6 +126,51 @@ def matrix_power(input, count):
     return result
 
 
+def matrix_rank(input, tol=None, hermitian=False):
+    """
+    Return matrix rank of array
+    Rank of the array is the number of singular values of the array that are
+    greater than `tol`.
+
+    Parameters
+    ----------
+    M : {(M,), (..., M, N)} array_like
+        Input vector or stack of matrices.
+    tol : (...) array_like, float, optional
+        Threshold below which SVD values are considered zero. If `tol` is
+        None, and ``S`` is an array with singular values for `M`, and
+        ``eps`` is the epsilon value for datatype of ``S``, then `tol` is
+        set to ``S.max() * max(M.shape) * eps``.
+    hermitian : bool, optional
+        If True, `M` is assumed to be Hermitian (symmetric if real-valued),
+        enabling a more efficient method for finding singular values.
+        Defaults to False.
+    Returns
+    -------
+    rank : (...) array_like
+        Rank of M.
+
+    """
+
+    is_input_dparray = isinstance(input, dparray)
+
+    if not use_origin_backend(input) and is_input_dparray:
```

Comment:
It looks like you could skip checks from default values by using kwargs
https://github.com/IntelPython/dpnp/blob/master/dpnp/dpnp_iface_mathematical.py#L142